pm_nq/pm_nq.xml

```diff
@@ -0,0 +1,232 @@
+<tool id="pm_nq" name="PyMuonSuite Quantum Averaging" version="@TOOL_VERSION@+galaxy@WRAPPER_VERSION@" python_template_version="3.5" profile="22.05">
+    <description>generate and average displaced structures</description>
+    <macros>
+        <!-- version of underlying tool (PEP 440) -->
+        <token name="@TOOL_VERSION@">0.2.1</token>
+        <!-- version of this tool wrapper (integer) -->
+        <token name="@WRAPPER_VERSION@">0</token>
+        <!-- citation should be updated with every underlying tool version -->
+        <!-- typical fields to update are version, month, year, and doi -->
+        <token name="@TOOL_CITATION@">
+            @software{pymuon-suite,
+                author = {Sturniolo, Simone and Liborio, Leandro and Chadwick, Eli and Murgatroyd, Laura and Laverack, Adam and {Muon Spectroscopy Computational Project}},
+                license = {GPL-3.0},
+                title = {{pymuon-suite}},
+                url = {https://github.com/muon-spectroscopy-computational-project/pymuon-suite},
+                version = {v0.2.1},
+                month = {2},
+                year = {2022},
+                doi = {}
+            }
+        </token>
+        <import>muon_macros.xml</import>
+    </macros>
+    <creator>
+        <person givenName="Patrick" familyName="Austin" url="https://github.com/patrick-austin"/>
+        <organization url="https://muon-spectroscopy-computational-project.github.io/index.html" name="The Muon Spectroscopy Computational Project"/>
+    </creator>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">pymuonsuite</requirement>
+        <requirement type="package" version="1.0.0">euphonic</requirement>
+        <requirement type="package" version="3.0">zip</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        structure_name=\$(sed 's/ //g' <<< '$structure.name') &&
+        #if $task.task == 'w':
```

Comment:
@patrick-austin can you maybe give me a small introduction why this name-wrangling is needed? What do you need to archive here?

Reply:
Before my involved in the project, there was an issue to remove whitespace from file names using `sed` so that we could `ln -s` them #54. As we've adopted this pattern of remove the spacing and symlinking elsewhere, I've continued it in this tool as well.

I believe we can't just use the `/tmp/tmpenfof_cq/files/7/3/6/dataset_7368bdbc-979b-41d3-98b2-c4b3138090fc.dat` filepath as a lot of the tools we're using require that the various input files associated with a single sample (e.g. Si) differ only in their file extension:

- Si.cell
- Si.param
- Si.castep
- Si.phonon
- ...etc.
 
And so if we provided `/tmp/tmpenfof_cq/files/7/3/6/dataset_7368bdbc-979b-41d3-98b2-c4b3138090fc.dat` as the structure (.cell) file, it would expect to also find a file named `/tmp/tmpenfof_cq/files/7/3/6/dataset_7368bdbc-979b-41d3-98b2-c4b3138090fc.castep`  - which doesn't exist. So we symlink them to have consistent filenames so that the tools will run, which in turn requires we don't have spaces in the filename.